src/lib/common/sol-platform-detect.c

```diff
@@ -215,9 +217,14 @@ sol_platform_detect(void)
     struct sol_json_token token, key, value, platform_name = { NULL };
     enum sol_json_loop_reason reason;
 
-    json_doc = _json_open_doc(DATADIR "/platform_detect.json", &scanner);
-    if (!json_doc)
-        return NULL;
+    json_doc = _json_open_doc("/etc/" PLATFORM_JSON, &scanner);
```

Comment:
in this case I believe it's better to use /etc/soletta/ and then a name, instead of putting multiple soletta files in the /etc itself. With linux-micro we use PKGSYSCONFDIR, you should use that instead of "/etc" and it should evaluate to "/etc/soletta" (currently does not, a bug I just noticed right now -- @dorileo  please take a look)